airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/validator/RedshiftUtil.java

```diff
@@ -0,0 +1,29 @@
+package io.airbyte.integrations.destination.redshift.validator;
+
+import static io.airbyte.integrations.destination.redshift.constants.RedshiftDestinationConstants.UPLOADING_METHOD;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * Helper class for Destination Redshift connector.
+ */
+public class RedshiftUtil {
+
+  private RedshiftUtil() {
+  }
+
+  public static JsonNode doesJsonNodeContainUploadingMethod(final JsonNode config) {
+    return config.has(UPLOADING_METHOD) ? config.get(UPLOADING_METHOD) : config;
```

Comment:
is the idea here that some destinations put the S3 config options directly into the root spec?

If yes, I think it would be better to rename this method. Maybe something like `findS3Options`, to make it clear what's being returned (the current name implies that it returns a boolean)

Reply:
@edgao this flow was done for backward compatibility. Any previous spec contains config in the root of the node.